doc/specification/language-proposal.md

```diff
@@ -1204,45 +1225,55 @@ regime A {
+Any `when` conditions or `on` conditions defined in an outer regime, including the topmost unnamed regime, are inherited by inner regimes, are tested before any additional conditions defined in the inner regime are considered.
+
+#### When and on semantics
 
-#### When semantics
+When clauses are triggered when the mechanism state satisfies the corresponding predicate, and an on-clause is triggered when an event of the appropriate type is delivered to the mechanism. The state is updated according to the `state = ` clause, and if a `regime` clause is present, the dynamics are shifted to the given regime.
 
-When clauses are triggered when the mechanism state satisfies the corresponding predicate, or when an event of the appropriate type is delivered to the mechanism. The state is updated according to the `state = ` clause, and if a `regime` clause is present, the dynamics are shifted to the given regime.
+Associated when- and on-clauses are considered first by scope — conditions defined in outer regimes take priority over conditions defined in an inner regime — and then by order of definition. If more than one when- or on-clause applies, they are applied in order of definition. Note that the effect of a clause may cause a transition to a regime where the set of applicable clauses differs, whence the new set is considered in its stead.
 
-A given when clause associated with a regime _R_, with predicate _p_ applies at the point in time _t₁_ if the dynamics are in regime _R_ at _t₁_, _p_(_t₁_) is true and either _p_(_t_) is false or dynamics were in a regime _R'_ without this when clause for _t_ ∈ (_t₁_-δ, _t₁_) for some δ>0. All predicates are considered false for _t_ &lt; 0.
+A given when clause associated with a regime _R_, with predicate _p_ applies at the point in time _t₁_ if the dynamics are in regime _R_ at _t₁_, _p_(_t₁_) is true and either _p_(_t_) is false or dynamics were in a different regime _R'_ not associated with the when clause for _t_ ∈ (_t₁_-δ, _t₁_) for some δ>0. All predicates are considered false for _t_ &lt; 0. In addition, an applicable when clause can be satisfied at most once for any point in time _t₁_.
 
-If more than one when clause applies, they are applied in order of definition. Note that the effect of a when clause may cause a transition to a regime where the set of applicable when-clauses differs, whence the new set is considered in its stead. In any instance, no single when clause may be triggered more than once by the same event or, for predicate-based when clauses, at the same time.
+If events _e₁_, _e₂_, …, _eₙ_ are delivered at time _t_, applicable on-clauses are considered in turn for _e₁_, performing any state mutations or regime transistions as required (which may change in turn the set of applicable on-clauses), then for _e₂_, and so on. Each event can trigger at most one on-clause. No when-clauses are considered at time _t_ until all events delivered at _t_ are considered.
```

Comment:
> No when-clauses are considered at time _t_ until all events delivered at _t_ are considered.

So would the following (Same system as below, with statement D moved up), have the same analysis as the system described below?

```
initial regime X state = 0;
regime X {
    on weight = event; when weight>2 state = 1;    # (A)
    when state == 4 regime = Y; state = 5;         # (D)
    on weight = event; regime = Y; state = 2;      # (B)
    on weight = event; state = 3;                  # (C)
}
regime Y {
    when true regime = X; state = 4;               # (E)
}
```

Reply:
That's the intent, yes.